src/rgw/rgw_bucket.h

```diff
@@ -34,7 +34,8 @@ extern int rgw_bucket_parse_bucket_instance(const string& bucket_instance, strin
 extern int rgw_bucket_parse_bucket_key(CephContext *cct, const string& key,
                                        rgw_bucket* bucket, int *shard_id);
 
-extern int rgw_bucket_instance_remove_entry(RGWRados *store, string& entry, RGWObjVersionTracker *objv_tracker);
+extern int rgw_bucket_instance_remove_entry(RGWRados *store, const string&& entry,
```

Comment:
should either be `string&&` or `const string&`. constructing another string with a `const string&&` will call its copy constructor instead of the move constructor. i set up an example at https://www.godbolt.org/z/IdHVRX, where you can switch between `#if 1` and `#if 0` to see the difference

but since `RGWMetadataManager::remove_entry()` takes `const string&` anyway, this should also. it's fine to pass an rvalue like `bucket.get_key()` to a function taking const ref - it just doesn't work for non-const ref

Reply:
I assume the use of const string& is more natural here, anyway

Reply:
My mistake. Double fix when the first was sufficient.